pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kubelet/templates/scripts/health-monitor.tpl.sh

```diff
@@ -46,8 +46,18 @@ function kubelet_monitoring {
       continue
     fi
 
-    node_object="$(kubectl get nodes -l kubernetes.io/hostname=$(hostname) -o json)"
-    node_status="$(echo $node_object | jq -r '.items[0].status')"
+    node_name=
+    if [[ -s "{{ .pathNodeName }}" ]]; then
+      node_name="$(cat "{{ .pathNodeName }}")"
+    fi
+    if [[ -z "$node_name" ]]; then
+      echo "Node name is not known yet, waiting..."
+      sleep 20
+      continue
+    fi
+
+    node_object="$(kubectl get nodes "$node_name" -o json)"
```

Comment:
Nit: just to make it obvious that we have a request for a single Node.

```suggestion
    node_object="$(kubectl get node "$node_name" -o json)"
```